wodin_deploy/wodin_constellation.py

```diff
@@ -0,0 +1,117 @@
+from constellation import Constellation, ConstellationContainer, ConstellationMount, docker_util
+
+
+class WodinConstellation:
+    def __init__(self, cfg):
+        self.cfg = cfg
+        redis = redis_container(cfg)
+        odin_api = odin_api_container(cfg)
+        sites = sites_containers(cfg)
+        proxy = wodin_proxy_container(cfg)
+
+        containers = [redis, odin_api, *sites, proxy]
+
+        self.obj = Constellation(
+            "wodin",
+            cfg.container_prefix,
+            containers,
+            cfg.network,
+            {"redis-data": "redis-data", "wodin-config": "wodin-config"},
+            data=cfg,
+        )
+
+    def start(self, **kwargs):
+        self.obj.start(**kwargs)
+
+    def stop(self, **kwargs):
+        self.obj.stop(**kwargs)
+
+    def status(self):
+        self.obj.status()
+
+
+def redis_container(cfg):
+    redis = cfg.redis
+    redis_mounts = [ConstellationMount("redis-data", "/data")]
+    return ConstellationContainer(redis["name"], redis["ref"], mounts=redis_mounts, network=cfg.network)
+
+
+def odin_api_container(cfg):
+    odin_api = cfg.odin_api
+    return ConstellationContainer("api", odin_api["ref"], network=cfg.network)
+
+
+def get_wodin_container(cfg, site, site_dict):
+    wodin = cfg.wodin
+    prefix = cfg.container_prefix
+    wodin_mounts = [ConstellationMount("wodin-config", "/wodin/config")]
+    container_name = f"{wodin['name']}-{site}"
+
+    return ConstellationContainer(
+        container_name,
+        wodin["ref"],
+        mounts=wodin_mounts,
+        network=cfg.network,
+        entrypoint=[
+            "/wodin/docker/pull-site-and-start.sh",
+            site_dict["ref"],
+            site_dict["url"],
+            f"/wodin/config/{site}",
+            f"--redis-url=redis://{prefix}-{cfg.redis['name']}:6379",
+            f"--odin-api=http://{prefix}-api:8001",
+            f"--base-url=http://localhost/{site_dict['urlPath']}",
+            f"/wodin/config/{site}",
+        ],
+    )
+
+
+def sites_containers(cfg):
+    sites = cfg.sites
+    wodin_containers = []
+    for site in sites.keys():
+        site_dict = sites[site]
+        wodin_containers.append(get_wodin_container(cfg, site, site_dict))
+    return wodin_containers
+
+
+def create_index_page(sites):
+    html_strings = [
```

Comment:
it might be nice to use some html template instead of this at some point, but let's keep it simple for now!

Reply:
ahhh potentially, perhaps we can put a script into the wodin-proxy because if we were to create a html template inside wodin-deploy, doing `docker cp` is a real pain with docker-py